src/components/[guild]/Requirements/RoleRequirements.tsx

```diff
@@ -14,11 +14,18 @@ const VIRTUAL_LIST_REQUIREMENT_LIMIT = 10
 const PARENT_PADDING = "var(--chakra-space-5)"
 
 const RoleRequirements = ({ role }: Props) => {
-  const isVirtualList = role.requirements?.length > VIRTUAL_LIST_REQUIREMENT_LIMIT
-  const sliceIndex = (role.requirements?.length ?? 0) - 3
-  const shownRequirements = (role.requirements ?? []).slice(0, 3)
+  const requirements = role.hiddenRequirements
+    ? [
+        ...role.requirements,
+        { isHidden: true, type: "FREE" } as unknown as Requirement,
```

Comment:
We could maybe add the `isHidden?: boolean` prop to the `Requirement` type, and then we could just cast this object to `Requirement` type directly?